packages/sdk/src/types/primitiveTypes.ts

```diff
@@ -16,28 +16,78 @@ export enum PrimitiveShape {
 	Plane = 'plane'
 }
 
-/**
- * The size, shape, and description of a primitive.
- */
-export interface PrimitiveDefinition {
-	// TODO: Make this a discriminated union type.
+export type SpherePrimitiveDefinition = {
+	shape: PrimitiveShape.Sphere
+	/**
+	 * The bounding box size of the primitive.
+	 */
+	dimensions?: Partial<Vector3Like>;
+	/**
+	 * The number of horizontal or radial segments of spheres, cylinders, capsules, and planes.
+	 */
+	uSegments?: number;
 	/**
-	 * The general shape of the defined primitive.
+	 * The number of vertical or axial segments of spheres, capsules, and planes.
 	 */
-	shape: PrimitiveShape;
+	vSegments?: number;
+};
 
+export type BoxPrimitiveDefinition = {
+	shape: PrimitiveShape.Box
 	/**
 	 * The bounding box size of the primitive.
 	 */
 	dimensions?: Partial<Vector3Like>;
+};
 
+export type CapsulePrimitiveDefinition = {
+	shape: PrimitiveShape.Capsule
+	/**
+	 * The bounding box size of the primitive.
+	 */
+	dimensions?: Partial<Vector3Like>;
 	/**
 	 * The number of horizontal or radial segments of spheres, cylinders, capsules, and planes.
 	 */
 	uSegments?: number;
+	/**
+	 * The number of vertical or axial segments of spheres, capsules, and planes.
+	 */
+	vSegments?: number;
+};
 
+export type CylinderPrimitiveDefinition = {
+	shape: PrimitiveShape.Cylinder
+	/**
+	 * The bounding box size of the primitive.
+	 */
+	dimensions?: Partial<Vector3Like>;
+	/**
+	 * The number of horizontal or radial segments of spheres, cylinders, capsules, and planes.
+	 */
+	uSegments?: number;
+};
+
+export type PlanePrimitiveDefinition = {
+	shape: PrimitiveShape.Plane
+	/**
+	 * The bounding box size of the primitive.
+	 */
+	dimensions?: Partial<Vector3Like>;
+	/**
+	 * The number of horizontal or radial segments of spheres, cylinders, capsules, and planes.
+	 */
+	uSegments?: number;
 	/**
 	 * The number of vertical or axial segments of spheres, capsules, and planes.
 	 */
 	vSegments?: number;
-}
+};
+
+export type PrimitiveDefinition

```

Comment:
This is not a very friendly data structure, which is why the assets API doesn't use it. I would be in favor of leaving this unchanged as a network type, then having a separate discriminated union for the CreatePrimitive API. That way we could have a `radius` field for example, but still have a unified data structure for the network message.

Reply:
I like that approach. I was wondering why we didn't have a radius field.